src/systemctl/systemctl-list-units.c

```diff
@@ -932,3 +932,230 @@ int verb_list_automounts(int argc, char *argv[], void *userdata) {
+                else
+                        /* This should never happen, currently our path units can only trigger a
+                         * single unit. But let's handle this anyway, who knows what the future
+                         * brings? */
+                        r = table_add_cell(table, NULL, TABLE_STRV, p->triggered);
+                if (r < 0)
+                        return table_log_add_error(r);
+        }
+
+        r = output_table(table);
+        if (r < 0)
+                return r;
+
+        if (arg_legend != 0)
+                output_legend("path", ps->count);
+
+        return 0;
+}
+
+int verb_list_paths(int argc, char *argv[], void *userdata) {
+        _cleanup_(message_set_freep) Set *replies = NULL;
+        _cleanup_strv_free_ char **machines = NULL, **units = NULL;
+        _cleanup_free_ UnitInfo *unit_infos = NULL;
+        _cleanup_(path_infos_free) struct path_infos path_infos = {};
+        int r, n;
+        sd_bus *bus;
+
+        r = acquire_bus(BUS_MANAGER, &bus);
+        if (r < 0)
+                return r;
+
+        pager_open(arg_pager_flags);
+
+        r = expand_unit_names(bus, strv_skip(argv, 1), ".path", &units, NULL);
+        if (r < 0)
+                return r;
+
+        if (argc == 1 || units) {
+                n = get_unit_list_recursive(bus, units, &unit_infos, &replies, &machines);
+                if (n < 0)
+                        return n;
+
+                for (const UnitInfo *u = unit_infos; u < unit_infos + n; u++) {
+                        _cleanup_strv_free_ char **conditions = NULL, **paths = NULL, **triggered = NULL;
+                        int c;
+
+                        if (!endswith(u->id, ".path"))
+                                continue;
+
+                        r = get_triggered_units(bus, u->unit_path, &triggered);
+                        if (r < 0)
+                                return r;
+
+                        c = get_paths(bus, u->unit_path, &conditions, &paths);
+                        if (c < 0)
+                                return c;
+
+                        if (!GREEDY_REALLOC(path_infos.items, path_infos.count + c))
+                                return log_oom();
+
+                        for (int i = 0; i < c; i++)
+                                path_infos.items[path_infos.count + i] = (struct path_info) {
+                                        .machine = u->machine,
+                                        .id = u->id,
+                                        .condition = conditions[i],
+                                        .path = paths[i],
+                                        .triggered = triggered,
+                                        .own_triggered = i == 0,
+                                };
+
+                        conditions = mfree(conditions);
+                        paths = mfree(paths);
+                        triggered = NULL;
```

Comment:
hmm, you invalidate `triggered` here after adding it to all item objects you just created, hence it's now owned by which code specifically? i.e. it should have one owner who is responsible for freeing it.

Reply:
oh, now i get it, "own_triggered" indicates ownership. i see.

but hwat happens if c is 0?

Reply:
please find a different solution (copying with strdup()?), this is too "magic" for my taste

Reply:
I copied this from `verb_list_sockets` :-) (Granted, that code probably hasn't changed since it was written 10+ years ago. And the coding style has changed since these days, esp. around memory management...) I'll see what I can do to make it more palatable.